tasks/Functions-client/buildRequestJSON.js

```diff
@@ -0,0 +1,99 @@
+const { simulateRequest, buildRequest, getRequestConfig } = require("../../FunctionsSandboxLibrary")
+const { networkConfig } = require("../../network-config")
+const readline = require("readline-promise").default
+const axios = require("axios")
+const fs = require("fs")
+
+task("functions-build-request", "Creates a JSON file with Functions request parameters")
+  .addOptionalParam("output", "Output file name (defaults to Functions-request.json)")
+  .setAction(async (taskArgs, hre) => {
+    if (network.name === "hardhat") {
+      throw Error(
+        'This command cannot be used on a local development chain.  Specify a valid network or simulate an Functions request locally with "npx hardhat functions-simulate".'
+      )
+    }
+
+    const OracleFactory = await ethers.getContractFactory("FunctionsOracle")
+    const oracle = await OracleFactory.attach(networkConfig[network.name]["functionsOracle"])
+
+    console.log("Simulating Functions request locally...")
+    const unvalidatedRequestConfig = require("../../Functions-request-config.js")
+    const requestConfig = getRequestConfig(unvalidatedRequestConfig)
+
+    if (requestConfig.secretsLocation === 1) {
+      if (!requestConfig.secrets || Object.keys(requestConfig.secrets).length === 0) {
+        requestConfig.secrets = requestConfig.perNodeSecrets[0] ?? {}
+      }
+      // Get node addresses for off-chain secrets
+      const [ nodeAddresses, publicKeys ] = await oracle.getAllNodePublicKeys()
+      if (requestConfig.secretsURLs && requestConfig.secretsURLs.length > 0) {
+        verifyOffchainSecrets(requestConfig.secretsURLs, nodeAddresses)
+      }
+    }
+
+    const { success, resultLog } = await simulateRequest(requestConfig)
+    console.log(`\n${resultLog}`)
+
+    // If the simulated JavaScript source code contains an error, confirm the user still wants to continue
+    if (!success) {
+      const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+      })
+      const q1answer = await rl.questionAsync(
+        "There was an error when running the JavaScript source code for the request.\nContinue? (y) Yes / (n) No\n"
+      )
+      rl.close()
+      if (q1answer.toLowerCase() !== "y" && q1answer.toLowerCase() !== "yes") {
+        return
+      }
+    }
+
+    // Fetch the DON public key from on-chain
+    const DONPublicKey = await oracle.getDONPublicKey()
+    // Remove the preceding 0x from the DON public key
+    requestConfig.DONPublicKey = DONPublicKey.slice(2)
+    // Build the parameters to make a request from the client contract
+    const request = await buildRequest(requestConfig)
+
+    fs.writeFileSync(taskArgs.output ?? 'Functions-request.json', JSON.stringify(request))
+    console.log(`Wrote request to ${taskArgs.output ?? 'Functions-request.json'}`)
+  })
+
+
+const verifyOffchainSecrets = async (secretsURLs, nodeAddresses) => {
```

Comment:
Are you planning to call this one from request.js?

Reply:
Yes, however this must be addressed after the other PR is merged since both PRs require this logic but they have now been split into separate PRs.

Reply:
Realistically, a lot of re-used logic can be put into a common `utils` to prevent repetition (such as the gas limit overrides), but this should be a separate PR.